### DISCUSSION TITLE
Preview is not updating in windows WSL

### DISCUSSION THREAD

**Discussion by nexdep** [quarto-dev/quarto-cli#9202] [Q&A] [answered]
### Description

Hello 

I am running the quarto 1.4.552 with vscode running in WSL2 (in windows 11). The WSL2 OS is Ubuntu 22.04.4 LTS.

I have two issues:

The preview is not working, meaning I manage to show the preview but if I modify the qmd file there is no way to see the preview updated either by:

- click reload in the preview pane
- refresh in the browser
- with the autorefresh option in settings.

The second issue is that to launch the preview I need to previously run the  command:

export XDG_RUNTIME_DIR=$(mktemp -d)    

As it was commented in a previous discussion. I was wondering if there is a way to fix this issue.

Just wanted to highlight these issues, these are not showstopper but quite annoying in the day to day use. I assume that some of these issues are related to the WSL environment?

Thanks for the help

**Comment by mcanouil** [comment]
Anyway you can describe the steps to reproduce the first issue? As is (with only "preview is not working") there is nothing we can do.

regarding the second issue, could you clarify the context and the discussion you are referring to?

**Reply by nexdep** [reply, marked as answer]
thanks for the reply.

it could be reproduced just by changing an hello world of quarto like this:

```qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded Python code.
When you run the code, it will produce a message.
```

however after better looking I found that by addressing the second issue using this fix:
https://github.com/quarto-dev/quarto-cli/issues/4594#issuecomment-1619177667

also the first one is fixed!

**Comment by cderv** [comment]
> The preview is not working, meaning I manage to show the preview but if I modify the qmd file there is no way to see the preview updated

See about how to use VSCODE extension at: https://quarto.org/docs/tools/vscode.html#render-and-preview

Quarto VS CODE extension will not re-render automatically the preview for input .qmd file. From the doc. it will only reload lightweight change on resources.  

To automatically render a change you are doing on source file, you need to configure Render on Save behavior for the extension: https://quarto.org/docs/tools/vscode.html#render-on-save

Does this helps ?

> The second issue is that to launch the preview I need to previously run the command:
> export XDG_RUNTIME_DIR=$(mktemp -d)

Just so you know, I am running also windows and WSL and it works ok for me without doing this . I don't recall to have set this, and I think it is the default, but this env var is for me set to 
````
> echo $XDG_RUNTIME_DIR
/run/user/1000/
````

> As it was commented in a previous discussion. I was wondering if there is a way to fix this issue.

Can you link to the issue ?

**Reply by nexdep** [reply]
https://github.com/quarto-dev/quarto-cli/discussions/9202#discussioncomment-8931776
